testsupport/tiers/checks.go

```diff
@@ -88,19 +84,80 @@ func NewChecks(tier string) (TierChecks, error) {
 		return &appstudioTierChecks{tierName: appstudio}, nil
 
-	case test:
-		return &testTierChecks{tierName: test}, nil
+	case testtier:
+		return &testTierChecks{tierName: testtier}, nil
 
 	default:
-		return nil, fmt.Errorf("no assertion implementation found for %s", tier)
+		return nil, fmt.Errorf("no assertion implementation found for %s", tier.Name)
 	}
 }
 
-type TierChecks interface {
-	GetNamespaceObjectChecks(nsType string) []namespaceObjectsCheck
-	GetClusterObjectChecks() []clusterObjectsCheck
-	GetExpectedTemplateRefs(hostAwait *wait.HostAwaitility) TemplateRefs
-	GetTierObjectChecks() []tierObjectCheck
+// NewChecksForCustomTier returns a `TierChecks` initialized with the tiers used in the CustomNSTemplateTier
+func NewChecksForCustomTier(tier *CustomNSTemplateTier) (TierChecks, error) {
+	c := &customTierChecks{}
+
+	// using the checks of the "cluster resources" NSTemplateTier
+	clusterChecks, err := NewChecksForTier(tier.ClusterResourcesTier)
+	if err != nil {
+		return nil, err
+	}
+	c.getClusterObjectChecks = clusterChecks.GetClusterObjectChecks
+
+	// using the checks of the "namespace resources" NSTemplateTier
+	namespaceChecks, err := NewChecksForTier(tier.ClusterResourcesTier)
+	if err != nil {
+		return nil, err
+	}
+	c.getNamespaceObjectChecks = namespaceChecks.GetNamespaceObjectChecks
+
+	// using default/wrapped tier for the deactivation timeout
+	c.getTierObjectChecks = func() []tierObjectCheck {
+		return []tierObjectCheck{
+			nsTemplateTier(tier.Name, tier.Spec.DeactivationTimeoutDays),
+		}
+	}
+
+	var clusterResourcesTmplRef *string
+	if tier.NSTemplateTier.Spec.ClusterResources != nil {
+		clusterResourcesTmplRef = &tier.NSTemplateTier.Spec.ClusterResources.TemplateRef
+	}
+	namespaceTmplRefs := make([]string, len(tier.NSTemplateTier.Spec.Namespaces))
+	for i, ns := range tier.NSTemplateTier.Spec.Namespaces {
+		namespaceTmplRefs[i] = ns.TemplateRef
+	}
+
+	c.getExpectedTemplateRefs = func(_ *wait.HostAwaitility) TemplateRefs {
+		return TemplateRefs{
+			ClusterResources: clusterResourcesTmplRef,
+			Namespaces:       namespaceTmplRefs,
+		}
+	}
+	return c, nil
+}
+
+var _ TierChecks = &customTierChecks{}
+
+type customTierChecks struct {
+	getClusterObjectChecks   func() []clusterObjectsCheck
+	getExpectedTemplateRefs  func(hostAwait *wait.HostAwaitility) TemplateRefs
+	getNamespaceObjectChecks func(nsType string) []namespaceObjectsCheck
+	getTierObjectChecks      func() []tierObjectCheck
+}
+
+func (c *customTierChecks) GetNamespaceObjectChecks(nsType string) []namespaceObjectsCheck {
+	return c.getNamespaceObjectChecks(nsType)
+}
+
+func (c *customTierChecks) GetClusterObjectChecks() []clusterObjectsCheck {
+	return c.getClusterObjectChecks()
+}
+
+func (c *customTierChecks) GetExpectedTemplateRefs(hostAwait *wait.HostAwaitility) TemplateRefs {
+	return c.getExpectedTemplateRefs(hostAwait)
+}
+
+func (c *customTierChecks) GetTierObjectChecks() []tierObjectCheck {
+	return c.getTierObjectChecks()
 }
```

Comment:
in your proposal above, the `customTierChecks` wraps `t *testing.T` which I'm not a huge fan of. Instead, I would rather return an error when calling `NewChecksForCustomTier`, as the current PR proposes. But we can discuss about (I know we wrap `t *testing.T` in other places)

Reply:
I proposed it because it would be consistent with the other checks (and I thought that you always opt for consistency).

As for wrapping `t *testing.T` - we do that all over the code. And in this particular case, it's totally fine because you instantiate TierCheck per one test - it's not shared between tests. It's something like an assertion type, and we do wrap ` t *testing.T` in the assertion types as well.

Reply:
yeah, I was referring to consistency with regards to not embedding `testing.T` in the `TierChecks` implementation ;) 
but it's fine, let's not argue and waste time on this and let me apply your suggestion :)

Reply:
> It's something like an assertion type, and we do wrap ` t *testing.T` in the assertion types as well.

that's a fair and valid point, indeed!

Reply:
done in 1226c6d